Editor/Scripts/MeshSyncServerInspector.cs

```diff
@@ -87,14 +88,20 @@ public void DrawServerSettings(MeshSyncServer t)
         }
     }
 
-    private void DrawInstanceSettings(MeshSyncServer t)
-    {
+    private void DrawInstanceSettings(MeshSyncServer t) {
         var style = EditorStyles.foldout;
-        style.fontStyle = FontStyle.Bold;
+        style.fontStyle        = FontStyle.Bold;
         t.foldInstanceSettings = EditorGUILayout.Foldout(t.foldInstanceSettings, "Instances", true, style);
-        if (t.foldInstanceSettings)
-        {
-            t.InstanceHandling = (BaseMeshSync.InstanceHandlingType)EditorGUILayout.EnumPopup("Instance handling", t.InstanceHandling);
+        if (t.foldInstanceSettings) {
+            var newInstanceHandling =
+                (BaseMeshSync.InstanceHandlingType)EditorGUILayout.EnumPopup("Instance handling", t.InstanceHandling);
+
+            if (t.InstanceHandling != newInstanceHandling &&
+                EditorUtility.DisplayDialog("Warning",
+                    "Changing the instance handling mode will delete any prefabs and previously synced objects for this server. Are you sure you want to do this?",
+                    "Yes", "No", DialogOptOutDecisionType.ForThisMachine, OPT_OUT_INSTANCE_HANDLING)) {
```

Comment:
Do you intent to use the opt out for the machine instead of the session?

Reply:
I think it makes sense for the machine, no?

Reply:
I think both make sense, but if I have to choose, I'll choose the session one, because there is no direct way to reset it back after saving it in EditorPrefs, which happens if we use `DialogOptOutDecisionType.ForThisMachine`.

Another alternative would be to add an option to reset this in MeshSync's preferences page.